packages/shared/src/entities/activity-feed/activity.interface.ts

```diff
@@ -0,0 +1,30 @@
+import { ChannelTypeEnum } from '../../types';
+import { IExecutionDetail } from '../execution-details';
+import { IJob as IJobBase } from '../job';
+import { INotificationTemplate } from '../notification-template';
+import { ISubscriber } from '../subscriber';
+
+export interface IActivityJob extends IJobBase {
+  executionDetails: IExecutionDetail[];
+}
+
+export interface IActivity {
+  _id: string;
+  _templateId: string;
+  _environmentId: string;
+  _organizationId: string;
+  _subscriberId: string;
+  _digestedNotificationId?: string;
+  transactionId: string;
+  channels: ChannelTypeEnum[];
+  to: {
+    subscriberId: string;
```

Comment:
What if this is a topic? Is the nesting necessary at this point or `to: string` is enough?

Reply:
Even for a topic, this data will always be the result of a fanout for a given subscriber. This aligns with the API result schema, so I left it as-is.